src/block.ts

```diff
@@ -0,0 +1,380 @@
+    switch (block.type) {
+
+        case ElementType.TEXT:
+            return new Ok({ kind: ElementType.TEXT, doc: docParser(block.textTypeData.html) });
+
+        case ElementType.IMAGE:
+
+            const masterAsset = block.assets.find(asset => asset.typeData.isMaster);
+            const { alt, caption, displayCredit, credit } = block.imageTypeData;
+            const imageBlock: Option<Result<string, Block>> = fromNullable(masterAsset)
+                .map(asset => asset.file)
+                .map(file => new Ok({
+                    kind: ElementType.IMAGE,
+                    alt,
+                    caption,
+                    displayCredit,
+                    credit,
+                    file,
+                }));
+
+            return imageBlock.withDefault(new Err('I couldn\'t find a master asset'));
+
+        case ElementType.PULLQUOTE:
+
+            const { html: quote, attribution } = block.pullquoteTypeData;
+            return new Ok({
+                kind: ElementType.PULLQUOTE,
+                quote,
+                attribution: fromNullable(attribution),
+            });
+
+        case ElementType.INTERACTIVE:
+            const { iframeUrl } = block.interactiveTypeData;
+            return new Ok({ kind: ElementType.INTERACTIVE, url: iframeUrl });
+
+        case ElementType.RICH_LINK:
+
+            const { url, linkText } = block.richLinkTypeData;
+            return new Ok({ kind: ElementType.RICH_LINK, url, linkText });
+
+        case ElementType.TWEET:
+            return tweetContent(block.tweetTypeData.id, docParser(block.textTypeData.html))
+                .map(content => ({ kind: ElementType.TWEET, content }));
+
+        default:
+            return new Err(`I'm afraid I don't understand the block I was given: ${block.type}`);
+    }
+}
+
+const parseAll = (docParser: DocParser) => (blocks: BlockElement[]): Result<string, Block>[] =>
+    blocks.map(parser(docParser));
+
+
+// ----- Renderer ----- //
+
+// The nodeType for ELEMENT_NODE has the value 1.
+function isElement(node: Node): node is Element {
+    return node.nodeType === 1;
+}
+
+function getAttrs(node: Node): React.AnchorHTMLAttributes<string> {
+    if (isElement(node)) {
+        return Array.from(node.attributes).reduce((attrs, attr) => {
+            return { ...attrs, [attr.name]: attr.value };
+        }, {});
+    } else {
+        return {};
+    }
+}
+
+const Paragraph = (props: { children?: ReactNode }): ReactElement =>
+    h('p', null, props.children);
+
+const anchorStyles = (colour: string): SerializedStyles => css`
+    color: ${colour};
+`;
+
+const Anchor = (props: { href: string; text: string; pillar: Pillar }): ReactElement =>
+    styledH(
+        'a',
+        { css: anchorStyles(getPillarStyles(props.pillar).kicker), href: props.href },
+        props.text,
+    );
+
+const bulletStyles = (colour: string): SerializedStyles => css`
```

Comment:
Won't this be needed outside of `blocks` to? e.g. standfirsts

Reply:
Probably. The location of some of this stuff will likely change. It's just in the `blocks` file for now because it's only used for that at the moment.